### PR TITLE
[candi] update containerd version to 1.6.14

### PR DESCRIPTION
**Pull request by RomanenkoDenys**
Signed-off-by: Denis Romanenko <denis.romanenko@flant.com>

## Description

This PR updates containerd to version 1.6.14 and removes containerd-fe due to needed patches already included in containerd release.
Also added deckhouse release requirement about minimal ubuntu os version in pr https://github.com/deckhouse/deckhouse/pull/3431, which may be merged first.

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: feature
summary: | 
Updated containerd version to 1.6.14.
Added deckhouse release requirement about minimal ubuntu os version.
impact: all containerd daemons should be restarted.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

File: modules/040-node-manager/requirements/check.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+func init() {
+	checkRequirementFunc := func(requirementValue string, getter requirements.ValueGetter) (bool, error) {
+		desiredVersion, err := semver.NewVersion(requirementValue)
+		if err != nil {
+			return false, err
+		}
+		currentVersionRaw, exists := getter.Get(minVersionValuesKey)
+		if !exists {
+			// no IngressNginxController CRs exist
```
Comment (yalosev) [inline modules/040-node-manager/requirements/check.go:40]: copy-paste magic %)

File: modules/040-node-manager/requirements/check.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import (
+	"errors"
+
+	"github.com/Masterminds/semver/v3"
+
+	"github.com/deckhouse/deckhouse/go_lib/dependency/requirements"
+)
+
+const (
+	minVersionValuesKey = "nodes:minimalOSVersion:ubuntu"
```
Comment (yalosev) [inline modules/040-node-manager/requirements/check.go:28]: `minUbuntuVersionValuesKey` ?

Comment (RomanenkoDenys) [reply modules/040-node-manager/requirements/check.go:28]: yep

File: modules/040-node-manager/requirements/check.go (hunk 1 above)
Comment (RomanenkoDenys) [reply modules/040-node-manager/requirements/check.go]: fixed

File: candi/bashible/bundles/centos/node-group/031_install_containerd.sh.tpl [hunk 3]
```diff
@@ -19,7 +19,7 @@ post-install() {
   systemctl daemon-reload
   systemctl enable containerd.service
 {{ if ne .runType "ImageBuilding" -}}
-  systemctl restart containerd.service
+  bb-flag-set containerd-need-restart
```
Comment (shvgn) [inline candi/bashible/bundles/centos/node-group/031_install_containerd.sh.tpl:22]: Nitpicking: `needs`

File: modules/040-node-manager/hooks/minimal_node_os_version_test.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+status:
+  nodeInfo:
+    osImage: CentOS Linux 7 (Core)
+`
+)
+
+var _ = Describe("node-manager :: minimal_node_os_version ", func() {
+	f := HookExecutionConfigInit(`{}`, `{}`)
+
+	Context("Nodes objects is not find", func() {
```
Comment (shvgn) [inline modules/040-node-manager/hooks/minimal_node_os_version_test.go:67]:
```suggestion
	Context("Node objects are not found", func() {
```

File: modules/040-node-manager/hooks/minimal_node_os_version_test.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+			f.RunHook()
+		})
+
+		It("Should have minimal version", func() {
+			Expect(f).To(ExecuteSuccessfully())
+			value, exists := requirements.GetValue(minVersionValuesKey)
+			Expect(exists).To(BeTrue())
+			Expect(value).To(BeEquivalentTo("18.4.5"))
+		})
+	})
```
Comment (shvgn) [inline modules/040-node-manager/hooks/minimal_node_os_version_test.go:133]: Please, add two more tests with unspecified OS. 

1. the value is not set when nodes do not contain OS info
2. the was initially set and is removed when nodes do not contain OS info

**Review by shvgn — CHANGES_REQUESTED** [review]

File: candi/bashible/bundles/centos/node-group/031_install_containerd.sh.tpl (hunk 3 above)
Comment (RomanenkoDenys) [reply candi/bashible/bundles/centos/node-group/031_install_containerd.sh.tpl:22]: I don't think it matters because it's an internal flag

Comment (RomanenkoDenys) [reply candi/bashible/bundles/centos/node-group/031_install_containerd.sh.tpl:22]: i think this is not needed

File: modules/040-node-manager/hooks/minimal_node_os_version_test.go (hunk 5 above)
Comment (RomanenkoDenys) [reply modules/040-node-manager/hooks/minimal_node_os_version_test.go]: done in https://github.com/deckhouse/deckhouse/pull/3431

File: modules/040-node-manager/hooks/minimal_node_os_version_test.go (hunk 4 above)
Comment (RomanenkoDenys) [reply modules/040-node-manager/hooks/minimal_node_os_version_test.go]: https://github.com/deckhouse/deckhouse/pull/3431

**Review by diafour — APPROVED** [review]

**Review by shvgn — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Review by yalosev — APPROVED** [review]

**Review by shvgn — APPROVED** [review]

**Pull request merged**